data/changelog/dune/2024-04-03-dune.3.15.0.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Dune 3.15.0
+tags: [dune, platform]
+changelog: |
+   ### Added
+   
+   - Add link flags to to `ocamlmklib` for ctypes stubs (#8784, @frejsoya)
+   
+   - Remove some unnecessary limitations in the expansions of percent forms in
+     install stanza. For example, the `%{env:..}` form can be used to select files
+     to be installed. (#10160, @rgrinberg)
+   
+   - Allow artifact expansion percent forms (`%{cma:..}`, `%{cmo:..}`, etc.) in
+     more contexts. Previously, they would be randomly forbidden in some fields.
+     (#10169, @rgrinberg)
+   
+   - Allow `%{inline_tests}` in more contexts (#10191, @rgrinberg)
+   
+   - Remove limitations on percent forms in the `(enabled_if ..)` field of
+     libraries (#10250, @rgrinberg)
+   
+   - Support dialects in `dune describe pp` (#10283, @emillon)
+   
+   - Allow defining executables or melange emit stanzas with the same name in the
+     same folder under different contexts. (#10220, @rgrinberg, @jchavarri)
+   
+   ### Fixed
+   
+   - coq: Delay Coq rule setup checks so OCaml-only packages can build in hybrid
+     Coq/OCaml projects when `coqc` is not present. Thanks to @vzaliva for the
+     test case and report (#9845, fixes #9818, @rgrinberg, @ejgallego)
+   
+   - Fix conditional source selection with `select` on `bigarray` in OCaml 5
+     (#10011, @moyodiallo)
+   
+   - melange: fix inconsistency in virtual library implementation. Concrete
+     modules within a virtual library can now refer to its virtual modules too
+     (#10051, fixes #7104, @anmonteiro)
+   
+   - melange: fix a bug that would cause stale `import` paths to be emitted when
+     moving source files within `(include_subdirs ..)` (#10286, fixes #9190,
+     @anmonteiro)
+   
+   - Dune file formatting: output utf8 if input is correctly encoded (#10113,
+     fixes #9728, @moyodiallo)
+   
+   - Fix expanding dependencies and locks specified in the cram stanza.
+     Previously, they would be installed in the context of the cram test, rather
+     than the cram stanza itself (#10165, @rgrinberg)
+   
+   - Fix bug with `dune exec --watch` where the working directory would always be
+     set to the project root rather than the directory where the command was run
+     (#10262, @gridbugs)
+   
+   - Regression fix: sign executables that are promoted into the source tree
+     (#10263, fixes #9272, @emillon)
+   
+   - Fix crash when decoding dune-package for libraries with `(include_subdirs
+     qualified)` (#10269, fixes #10264, @emillon)
+   
+   ### Changed
+    
+    - Remove the `--react-to-insignificant-changes` option. (#10083, @rgrinberg)
+---
+We're happy to announce that Dune 3.15.0 is now available.
+This feature has many fixes and new features that you can find in the changelog.
+
+There are a few new features that we would like to specially highlight.
+
+### Removal of previous limitations in many forms
```

Comment:
can you add PR numbers and author handles?

Reply:
I've added them, rewriting them in some places so it would disrupt the text as little as possible.

Reply:
My suggestion was to use the same style as in previous announcements:

```markdown
### Removal of previous limitations in many forms (#xxxx, @yyyy)
```

I think it's sufficient and less disruptive, so you can do as you prefer.